File: Real-estate-market-IB190099.Services/Database/Advertise.cs
using System;
using System.Collections.Generic;

namespace Real_estate_market_IB190099.Services.Database;

public partial class Advertise
{
    public int Id { get; set; }

    public string? Status { get; set; }

    public string? Type { get; set; }

    public int? PropertyId { get; set; }

    public int? UserId { get; set; }

    public DateTime? DateCreated { get; set; }

    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();

    public virtual Property? Property { get; set; }

    public virtual ICollection<SavedAdvertise> SavedAdvertises { get; set; } = new List<SavedAdvertise>();

    public virtual User? User { get; set; }
}

File: Real-estate-market-IB190099.mobile/realestatemobile/lib/model/advertise.dart
// ignore_for_file: non_constant_identifier_names

import 'dart:core';
import 'dart:ffi';
import 'package:json_annotation/json_annotation.dart';
import 'package:realestatemobile/model/property.dart';
import 'package:realestatemobile/model/user.dart';

part 'advertise.g.dart';

@JsonSerializable()
class Advertise {
  Advertise() {}
  int? id;
  String? status;
  String? type;
  int? propertyId;
  int? userId;
  DateTime? dateCreated;
  Property? property;
  User? user;
  int? employeeId;

  factory Advertise.fromJson(Map<String, dynamic> json) =>
      _$AdvertiseFromJson(json);

  /// Connect the generated [_$AdvertiseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$AdvertiseToJson(this);
}
